dspace-api/src/main/java/org/dspace/authenticate/AuthenticationMethod.java

```diff
@@ -224,4 +224,14 @@ public String loginPageURL(Context context,
      * @return whether the authentication method is being used.
      */
     public boolean isUsed(Context context, HttpServletRequest request);
+
+    /**
+     * Check if the given challenge is valid to change the password of the given
+     * ePerson
+     * @param  context   The DSpace context
+     * @param  ePerson   the ePerson related to the password change
+     * @param  challenge The challenge to check
```

Comment:
@LucaGiamminonni : I haven't had a chance to do a full code review yet.  But, I wanted to quickly note that I find the term `challenge` here to be somewhat confusing.  Normally, I think of an authentication `challenge` more like a captcha or similar (along these lines: https://www.arkoselabs.com/explained/challenge-response-authentication/).

Is there a reason not to just call this the `current_password`?

Personally, I feel calling this param the "current_password" would make the code easier to understand, but maybe I'm misunderstanding why you use the term "challenge".

Reply:
@mwoodiupui : I agree with your points, but wanted to also make sure to ask if you have a better term to suggest (so that we can move quickly to refactor this code as needed)?  

Do you agree this code should be updated to simply use `current_password` instead of `challege`? Or did you have any  other ideas?

I think it's obviously important for us to quickly come up with a new term, so that the changes can be made across all three PRs (contract, REST impl, UI impl).

Reply:
I agree.  "Current password" is exactly what it is.